katsdpimageutils/primarybeamcorrection/primary_beam_correction.py

```diff
@@ -1,188 +1,219 @@
+    
+    Parameters
+    ----------
+    separation_rad: numpy array
+        Radial separation array
+    c_freq: numpy array
+        An array of central frequencies for each frequency plane
     """
     rho = separation_rad
     # Degrees to radians conversion
-    conv = np.pi / 180.
-    # converting arcminutes to radians
-    v_beam_rad = 89.5 / 60. * conv
-    h_beam_rad = 86.2 / 60. * conv
-    # Taking the Geometric mean for the veritical and horizontal cut through the beam.
-    vh_beam_mean = np.sqrt(v_beam_rad*h_beam_rad)
+    v_beam_rad = np.deg2rad(89.5 / 60.)
+    h_beam_rad = np.deg2rad(86.2 / 60.)
+    # Taking the Geometric mean for the vertical and horizontal cut through the beam.
+    vh_beam_mean = np.sqrt(v_beam_rad * h_beam_rad)
     flux_density = []
     for nu in c_freq:
+        # Converting GHz to Hz
         nu = nu/1.e9
-        theta_b = vh_beam_mean*(1./nu)
+        theta_b = vh_beam_mean / nu
         ratio = rho/theta_b
-        num = np.cos(1.189*np.pi*ratio)
-        dem = 1-4*(1.189*ratio)**2
-        a_b = (num/dem)**2
+        num = np.cos(1.189 * np.pi * ratio)
+        den = 1-4 * (1.189 * ratio)**2
+        a_b = (num/den)**2
         flux_density.append(a_b)
     return flux_density
 
 
 def beam_pattern(path):
-    """
-    Making beam pattern using the Cosine-squared power approximation from Mauch et al. (2020)
+    """Make beam pattern using the Cosine-squared power approximation
+    from Mauch et al. (2020).
+    
+    Parameters
+    ----------
+    input_file : str
+        FITS file
     """
     # Reading the fits file
     data = read_fits(path)
-    # Getting pixel indices
-    row, col = np.indices((data.shape[2], data.shape[3]))
-    row = np.ravel(row)
-    col = np.ravel(col)
+    # Getting pixel coordinates
+    pixcrd = np.indices((data.shape[2], data.shape[3]))
     # Getting the central frequency of the image header given.
     c_freq = central_freq(path)
     # Getting radial separation between sources and the phase centre
-    phase_center, image_wcs = get_positions(path)
-    # Calling the radial_offset function
-    separation_rad = radial_offset(phase_center, image_wcs, row, col)
-    # Calling the cosine_power_pattern function
+    phase_center, image_wcs = get_position(path)
+    separation_rad = radial_offset(phase_center, image_wcs, pixcrd)
     beam_list = cosine_power_pattern(separation_rad, c_freq)
-    return beam_list
+    return beam_list, data
 
 
-def mad(data):
-    """
-    Calculating median absolute deviation (MAD).
+def standard_deviation(data):
+    """Compute the median and the estimate of the standard deviation
+    based on the median absolute deviation (MAD).
     """
     MAD_TO_SD = 1.4826
     med = np.nanmedian(data)
     dev = np.abs(data - med)
     return med, MAD_TO_SD * np.nanmedian(dev)
 
 
-def standard_deviation(data):
+def inverse_variance(data):
+    """ Calculate the inverse variance. Reject pixels more than 5 sigma from
```

Comment:
Remove the space after the triple-quote, and put a paragraph break after the first sentence (so that it forms the summary).

Reply:
Done